### PR TITLE
Functional performance

### PR DESCRIPTION
**Pull request by galengorski**
This is a small change to the `lag_data` function in `it_functions`. I have changed the lag so that if an input matrix is `[X,Y]` where `X` is the source and `Y` is the sink, the new `lag_data` function will produce a matrix `M_lagged` that is `[X-T, Y, Y-T]`, where `T` is the time lag. The previous version produced a matrix `[X-T, Y, Y-1]`. I have seen both lagging methods in the literature, you're either conditioning on the immediate history of `Y` or it's history at the time step of interest, but we want `Y-T` here. The other change that I made was to add a function called `find_bounds`, which finds the bounds of the time series of interest using percentiles. This is used for removing outliers for analysis.

### CODE REVIEW

File: methods_exploration/it_functions.py [hunk 1]
```diff
@@ -291,21 +320,21 @@ def calc_it_metrics(M, Mswap, n_lags, nbins, calc_swap = True, alpha = 0.01):
...
         TE.append(TEtemp)
-        TEcrittemp = calcTE_crit(M, shift = i, nbins = nbins, ncores = 8, alpha = 0.01)
+        TEcrittemp = calcTE_crit(M, shift = i, nbins = nbins, ncores = 8, alpha = 0.05)
         TEcrit.append(TEcrittemp)
         
         if calc_swap:
             TEtempswap = calcTE(Mswap, shift = i, nbins = nbins)
             TEswap.append(TEtempswap)
-            TEcrittempswap = calcTE_crit(Mswap, shift = i, nbins = nbins, ncores = 8, alpha = 0.01)
+            TEcrittempswap = calcTE_crit(Mswap, shift = i, nbins = nbins, ncores = 8, alpha = 0.05)
```
Comment (jds485) [inline methods_exploration/it_functions.py:337]:
```suggestion
            TEcrittempswap = calcTE_crit(Mswap, shift = i, nbins = nbins, ncores = 8, alpha = alpha)
```

File: methods_exploration/it_functions.py [hunk 2]
```diff
@@ -62,6 +62,33 @@ def remove_seasonal_signal(self, sr):
...
+        lower bound value
+    float
+        upper bound value
+
+    '''
+    if (lower == None) & (upper == None):
+        return None, None
+    if lower == None:
+        return None, np.nanpercentile(data, upper)
+    if upper == None:
```
Comment (jds485) [inline methods_exploration/it_functions.py:89]:
```suggestion
    if upper is None:
```

**Review by jds485 — CHANGES_REQUESTED** [review]
This looks good! I have a comment and minor code suggestions

File: methods_exploration/it_functions.py [hunk 3]
```diff
@@ -291,21 +320,21 @@ def calc_it_metrics(M, Mswap, n_lags, nbins, calc_swap = True, alpha = 0.01):
         M_short =  M_lagged[~np.isnan(M_lagged).any(axis=1)]
         MItemp = calcMI(M_short[:,(0,1)], nbins)
         MI.append(MItemp)
-        MIcrittemp = calcMI_crit(M_short[:,(0,1)], nbins, ncores = 8, alpha = 0.01)
+        MIcrittemp = calcMI_crit(M_short[:,(0,1)], nbins, ncores = 8, alpha = 0.05)
```
[the remark this answers is not in the document]
Comment (galengorski) [reply methods_exploration/it_functions.py]: Added alpha an argument

File: methods_exploration/it_functions.py [hunk 4]
```diff
@@ -62,6 +62,33 @@ def remove_seasonal_signal(self, sr):
...
+
+    Returns
+    -------
+    float
+        lower bound value
+    float
+        upper bound value
+
+    '''
+    if (lower == None) & (upper == None):
```
[the remark this answers is not in the document]
Comment (galengorski) [reply methods_exploration/it_functions.py]: Changes made

**Comment by galengorski** [comment]
Thanks for the review @jds485, I have made the comments suggested. Your comment about the number `88` made me realize that this branch had a few changes to `run_model.py`, those are now included, but most of them, we had talked about before.

**Review by jds485 — APPROVED** [review]

**Pull request merged**